velox/vector/tests/VectorMakerTest.cpp

```diff
@@ -458,6 +459,55 @@ TEST_F(VectorMakerTest, arrayVectorFromJson) {
   }
 }
 
+TEST_F(VectorMakerTest, nestedArrayVectorFromJson) {
+  auto arrayVector = maker_.nestedArrayVectorFromJson<int32_t>({
+      "[[1, 2], [2, 3, 4], [null, 7]]",
+      "[[1, 3, 7, 9], []]",
+      "[]",
+      "[null]",
+      "[[1, 2, null], [], null]",
+  });
+
+  VectorPtr expectedVector = maker_.arrayVectorNullable<int32_t>({
+      {{1, 2}},
+      {{2, 3, 4}},
+      {{std::nullopt, 7}},
+      {{1, 3, 7, 9}},
+      {{}},
+      {{}},
+      {{std::nullopt}},
+      {{1, 2, std::nullopt}},
+      {{}},
+      std::nullopt,
+  });
+
+  VectorPtr elementsOfArrayVector = arrayVector->elements();
+
+  EXPECT_EQ(*arrayVector->elements()->type(), *ARRAY(INTEGER()));
+  EXPECT_TRUE(arrayVector->elements()->mayHaveNulls());
+
+  auto rowOfArrays = arrayVector->elements();
+  EXPECT_TRUE(rowOfArrays->mayHaveNulls());
+
+  // 5 rows in array of arrays
+  EXPECT_EQ(5, arrayVector->size());
+
+  EXPECT_EQ(3, arrayVector->sizeAt(0));
+  EXPECT_EQ(2, arrayVector->sizeAt(1));
+
+  // Empty inner array
+  EXPECT_EQ(1, arrayVector->sizeAt(2));
+  EXPECT_FALSE(arrayVector->isNullAt(2));
```

Comment:
Add a +ve test for isNullAt with a null element.

Also add a +ve isNullAt check for a null nested array element.